scipy/stats/_binom_test.py

```diff
@@ -0,0 +1,165 @@
+from collections import namedtuple
+import numpy as np
+from scipy.optimize import brentq
+from ._discrete_distns import binom
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+
+class BinomTestResult:
```

Comment:
I think we need a general `StatsTestResult` class. I adapted `OptimizeResult` [here](https://github.com/scipy/scipy/pull/12531/files#diff-cc307f50f4eb0f9917b55fdf1dbb9b8c) without much thought.